### ISSUE TITLE
E2E performance

### ISSUE THREAD

**Issue by rumusan** [arthurdouillard/incremental_learning.pytorch#8]
Thanks for your great repetition. Could you provide some configuration of E2E to get good performance? 

Besides, is there much difference between single fc and multi fc?

Thank you again!

**Comment by rumusan** [comment]
I can not run E2E with this code.
In e2e.py:
"
for i, ((_, idxes), inputs, targets) in enumerate(train_loader, start=1):
"
it seems the "train_loader" only have "inputs" and "targets".

I'd appreciate it if you could help me.

**Comment by arthurdouillard** [comment]
Indeed the code for e2e was not adapted since I changed data.py.
I don't store anymore indexes (to log previous predictions) as it was prone to bugs, but simply call the old model in ram.

Look at iCaRL for pointers: https://github.com/arthurdouillard/incremental_learning.pytorch/blob/master/inclearn/models/icarl.py#L135

My local git is more advanced that this repo, but I won't release right now. It needs cleaning, and I am doing some experiments that are not ready to be shared. Sorry.

Anyway I was not able to reach paper performance with E2E. Their data aug scheme + their regularization is quite tricky. Their gradient noise destroys my performances for examples.

I would advise you to look for easier & better paper. See Large Scale Incremental Learning from this year CVPR for example.

**Comment by rumusan** [comment]
Thank you so much.

**Comment by rumusan** [comment]
Thank you for your advice. 

In Figure 4 of "Large Scale Incremental Learning",  "Classifier without bias removal" gets [85%, 72%, 59%, 51%, 40%]. However, I can only get [86%, 62%, 47%, 35%, 24%] in my experiments.  Implementation Details are the same as the section 6.2. in the paper, except I implemented it with pytorch.

"Bias Correction Layer" does bring great improvement. However, I still can not get the performance presented in the paper. 

I think there may be some important points that I have not noticed.

Could you reproduce the results of the paper? Could you give me some help? 

Many thanks!

**Comment by hardik2396** [comment]
@rumusan  can you provide an implementation of large scale incremental learning? 

Thanks in advanced.
